src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -574,27 +578,71 @@ const StudentApplication = ({ userProfile, authentication }) => {
       } else {
         createSnackbar('Something went wrong while trying to save the application.', 'error');
       }
-      setSaving('failed');
+      setSaving('error');
     } finally {
       if (saveButtonAlertTimeout === null) {
         // Shows the success icon for 6 seconds and then returns back to normal button
         setSaveButtonAlertTimeout(
           setTimeout(() => {
             setSaveButtonAlertTimeout(null);
             setSaving(false);
-          }, 6000),
+          }, DYNAMIC_ICON_TIMEOUT),
         );
       }
       return result;
     }
   };
 
+  const handleDeleteAppAccepted = () => {
+    // The method is separated from callback because the housing API service must be handled inside an async method
+    deleteApartmentApplication();
+    handleCloseOkay();
+  };
+
+  /**
+   * Delete the current application in the database
+   *
+   * @async
+   * @function deleteApartmentApplication
+   */
+  const deleteApartmentApplication = async () => {
+    setDeleting(true);
+    setDeleteButtonAlertTimeout(null);
+    try {
+      const result = await housing.deleteApartmentApplication(applicationDetails.ApplicationID);
+      if (result) {
+        setDeleting('success');
+        loadApplication();
+        setApplicationCardsOpen(false);
+      } else {
+        throw new Error('Failed to delete application');
+      }
+    } catch (e) {
+      if (e instanceof AuthError) {
+        createSnackbar('You are not authorized to make changes to this application.', 'error');
```

Comment:
I got this error ("You are not auth...") whenever I would try to delete the application as a non-original editor.

Reply:
(Someone who was not originally the editor but is now cannot delete the application.)

Reply:
That is likely an API issue, since it is a 401 which you will get from the StateYourBusiness